### PR TITLE
Landing page

### PR DESCRIPTION
**Pull request by DinDin985**
WIP

### CODE REVIEW

File: apps/chameleon/components/ui/button.tsx [hunk 1]
```diff
@@ -17,6 +17,8 @@ const buttonVariants = cva(
         secondary:
           'bg-secondary text-secondary-foreground hover:bg-secondary/80',
         ghost: 'hover:bg-accent hover:text-accent-foreground',
+        aware: 'bg-white text-[#191919] hover:bg-[#e5e5e5]',
```
Comment (BBruington) [inline apps/chameleon/components/ui/button.tsx:20]: What is an aware button?

File: apps/chameleon/app/(auth)/register/page.tsx [hunk 2]
```diff
@@ -1,44 +1,138 @@
...
 import Link from 'next/link'
 
 import { cn } from '@/lib/utils'
 import { buttonVariants } from '@/components/ui/button'
-import { Icons } from '@/components/icons'
+import CarouselRegister from '@/components/signuppage/carouselregister'
+import TypeWritterEffect from '@/components/signuppage/typing'
 import { UserAuthForm } from '@/components/user-auth-form'
 
+import chameleon from '../../../public/icons/chameleon-logo-transparent-bg.png'
```
Comment (Balance8) [inline apps/chameleon/app/(auth)/register/page.tsx:11]: These imports need to be updated to use @/

File: apps/chameleon/app/(auth)/register/page.tsx [hunk 3]
```diff
@@ -1,44 +1,138 @@
...
 
 export default function RegisterPage() {
   return (
-    <div className="container grid h-screen w-screen flex-col items-center justify-center lg:max-w-none lg:grid-cols-2 lg:px-0">
+    <div className="relative container grid h-screen w-screen flex-col items-center justify-center lg:max-w-none lg:grid-cols-2 lg:px-0 animate-scroll bg-gradient-to-bl from-[#fe7489] via-[#eb8af9] to-[#08d3f9] bg-[length:250%_250%]">
       <Link
         href="/login"
         className={cn(
-          buttonVariants({ variant: 'ghost' }),
+          buttonVariants({ variant: 'aware' }),
```
Comment (Balance8) [inline apps/chameleon/app/(auth)/register/page.tsx:24]: lets not make a variant for this until its used more then 3 times

**Comment by DinDin985** [comment]
Note: Need to figure out the proper way to import images.

File: apps/chameleon/app/(auth)/register/page.tsx [hunk 4]
```diff
@@ -1,44 +1,138 @@
...
+          </CarouselContent>
+        </Carousel> */}
+      </div>
+
       <div className="lg:p-8">
-        <div className="mx-auto flex w-full flex-col justify-center space-y-6 sm:w-[350px]">
-          <div className="flex flex-col space-y-2 text-center">
-            <Icons.logo className="mx-auto h-6 w-6" />
-            <h1 className="text-2xl font-semibold tracking-tight">
+        <div className="mx-auto flex w-full flex-col justify-center space-y-6 sm:w-[350px] text-[#191919]">
```
Comment (Balance8) [inline apps/chameleon/app/(auth)/register/page.tsx:104]: we need to avoid text color arb values. these need to be brought into the design system color theme and then re written as text-insert-theme-color

File: apps/chameleon/app/global.css
Comment (Balance8) [inline apps/chameleon/app/global.css:1]: This does not need to be copied over, the global styles already exist at apps/chameleon/styles/globals.css

File: apps/chameleon/components/ui/button.tsx (hunk 1 above)
Comment (Balance8) [reply apps/chameleon/components/ui/button.tsx:20]: the name aware is not a clear variant name

File: apps/chameleon/components/ui/button.tsx [hunk 5]
```diff
@@ -17,6 +17,8 @@ const buttonVariants = cva(
...
+        signup: "",
```
Comment (Balance8) [inline apps/chameleon/components/ui/button.tsx:21]: remove this

File: apps/chameleon/components/ui/card.tsx [hunk 6]
```diff
@@ -9,7 +9,7 @@ const Card = React.forwardRef<
   <div
     ref={ref}
     className={cn(
-      'rounded-lg border bg-card text-card-foreground shadow-sm',
+      'h-full w-full border bg-card text-card-foreground shadow-sm',
```
Comment (Balance8) [inline apps/chameleon/components/ui/card.tsx:12]: dont edit the atom, if you need to extend this card into a unique new card, make a molecule that has the extensions that you need.

File: apps/chameleon/components/ui/carousel.tsx [hunk 7]
```diff
@@ -32,6 +33,18 @@ type CarouselContextProps = {
 
 const CarouselContext = React.createContext<CarouselContextProps | null>(null)
 
+const carouselVariant = cva({
```
Comment (Balance8) [inline apps/chameleon/components/ui/carousel.tsx:36]: this is not how we should handle extending the carousel.  You need to take the carousel atoms and make it into your own molecule that has the proper extensions.  See all the other files outside of the UI file.

File: apps/chameleon/components/ui/carousel.tsx [hunk 8]
```diff
@@ -157,12 +170,12 @@ const CarouselContent = React.forwardRef<
   const { carouselRef, orientation } = useCarousel()
 
   return (
-    <div ref={carouselRef} className="overflow-hidden">
+    <div ref={carouselRef} className="overflow-hidden h-full w-full">
```
Comment (Balance8) [inline apps/chameleon/components/ui/carousel.tsx:173]: dont edit the atom

File: apps/chameleon/components/user-auth-form.tsx [hunk 9]
```diff
@@ -30,6 +30,9 @@ export function UserAuthForm({ className, ...props }: UserAuthFormProps) {
   const [isLoading, setIsLoading] = React.useState<boolean>(false)
   const [isGitHubLoading, setIsGitHubLoading] = React.useState<boolean>(false)
   const searchParams = useSearchParams()
+  const variantType: string = props.styleVariant
```
Comment (Balance8) [inline apps/chameleon/components/user-auth-form.tsx:33]: Bringing in props like this breaks the headless model.  If you want to extend the current you need to do so with consideration for the atomic model.

File: apps/chameleon/components/user-auth-form.tsx [hunk 10]
```diff
@@ -80,27 +83,33 @@ export function UserAuthForm({ className, ...props }: UserAuthFormProps) {
...
             )}
             Sign In with Email
           </button>
         </div>
       </form>
+
       <div className="relative">
         <div className="absolute inset-0 flex items-center">
-          <span className="w-full border-t" />
+          {/* <span className="w-full border-t" /> */}
```
Comment (Balance8) [inline apps/chameleon/components/user-auth-form.tsx:101]: no comments in final product

File: apps/chameleon/components/user-auth-form.tsx [hunk 11]
```diff
@@ -80,27 +83,33 @@ export function UserAuthForm({ className, ...props }: UserAuthFormProps) {
...
         </div>
-        <div className="relative flex justify-center text-xs uppercase">
-          <span className="bg-background px-2 text-muted-foreground">
-            Or continue with
-          </span>
+        <div className="relative flex justify-center text-xs font-bold uppercase items-center">
+          <span className="w-[30%] border-t"></span>
```
Comment (Balance8) [inline apps/chameleon/components/user-auth-form.tsx:104]: dont use arb values if they can be avoided I am pretty sure there is a tailwind class for this

**Review by Balance8 — CHANGES_REQUESTED** [review]

File: apps/chameleon/app/(auth)/register/page.tsx [hunk 12]
```diff
@@ -1,44 +1,138 @@
 import { Suspense } from 'react'
+import Image from 'next/image'
 import Link from 'next/link'
 
 import { cn } from '@/lib/utils'
 import { buttonVariants } from '@/components/ui/button'
-import { Icons } from '@/components/icons'
+import CarouselRegister from '@/components/signuppage/carouselregister'
```
Comment (Balance8) [inline apps/chameleon/app/(auth)/register/page.tsx:7]: these imports never made it into the final git commit, you'll need to see that they get added

File: apps/chameleon/app/global.css
Comment (Balance8) [inline apps/chameleon/app/global.css:1]: this file needs to be removed. We already have a globals.css file located at apps/chameleon/styles/globals.css

File: apps/chameleon/app/(auth)/register/page.tsx (hunk 4 above)
Comment (DinDin985) [reply apps/chameleon/app/(auth)/register/page.tsx:104]: I added the arb values to the theming, thoughts on changing these values to the default black and white in order to help with eye-strain?
AddedColorsToThemeJPG

**Comment by DinDin985** [comment]
Added login page and converted carousel to responsive design.
LoginPage

File: apps/chameleon/components/ui/card.tsx [hunk 13]
```diff
@@ -1,6 +1,12 @@
-import * as React from 'react'
+import * as React from 'react';
+
+
+
+import { cn } from '@/lib/utils';
```
Comment (Balance8) [inline apps/chameleon/components/ui/card.tsx:5]: remove unused code

File: apps/chameleon/components/ui/carousel.tsx [hunk 14]
```diff
@@ -1,6 +1,7 @@
 'use client'
 
 import * as React from 'react'
+import { cva } from 'class-variance-authority'
```
Comment (Balance8) [inline apps/chameleon/components/ui/carousel.tsx:4]: remove unused code

File: apps/chameleon/components/ui/input.tsx [hunk 15]
```diff
@@ -10,10 +10,7 @@ const Input = React.forwardRef<HTMLInputElement, InputProps>(
     return (
       <input
         type={type}
-        className={cn(
-          'flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50',
-          className
-        )}
+        className={cn('flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50', className)}
```
Comment (Balance8) [inline apps/chameleon/components/ui/input.tsx:13]: revert this change

File: apps/chameleon/tailwind.config.js [hunk 16]
```diff
@@ -21,6 +21,8 @@ module.exports = {
     },
     extend: {
       colors: {
+        'dim-black': '#191919',
```
Comment (Balance8) [inline apps/chameleon/tailwind.config.js:24]: Let's remove this value in favor of a tailwind default black; find one on the colors page that is close, and remove this value from the code base. https://tailwindcss.com/docs/customizing-colors

File: apps/chameleon/tailwind.config.js [hunk 17]
```diff
@@ -21,6 +21,8 @@ module.exports = {
...
+        'neutral-200': '#e5e5e5',
```
Comment (Balance8) [inline apps/chameleon/tailwind.config.js:25]: this needs to be removed

File: apps/chameleon/app/(auth)/register/page.tsx (hunk 4 above)
Comment (Balance8) [reply apps/chameleon/app/(auth)/register/page.tsx]: Lets remove these and use tailwind default color theme

**Review by Balance8 — CHANGES_REQUESTED** [review]
@DinDin985  Fix these last few comments and remove the extra globals.css and fix the build errors on the CI/CD pipeline action and then we can merge 👍

**Comment by DinDin985** [comment]
Uploaded the images. But still failed to deploy open to any feedback.

**Review by Balance8 — APPROVED** [review]
Very well done! Huge milestone 🥂

**Comment by DinDin985** [comment]
Mark this day that I got my first major PR. DOPA DOWN!
